src/test/java/Pages/LoginPage.java

```diff
@@ -0,0 +1,46 @@
+package Pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class LoginPage extends BasePage {
+    public LoginPage(WebDriver givenDriver) {
+        super(givenDriver);
+    }
+
+    @FindBy(css = "input[type='email']")
+    WebElement emailField;
+    @FindBy(css = "input[type='password']")
+    WebElement passwordField;
+    @FindBy(css = "button[type=submit]")
+    WebElement submitBtn;
```

Comment:
You should use encapsulation of non-static and non-final fields.

If a variable is used only by other methods of that class, it should be declared as private.

I recommend:  make all the locators and WebElements private, because private keyword in java allows most restrictive access to variables and methods and offer strongest form of Encapsulation. 

For example

    ```
    private By shuffleBtnLocator = By.cssSelector(".btn-shuffle-all");
    private By firstSongLocator = By.cssSelector(".play");

```
   @FindBy(css = "img.avatar")
    private WebElement avatar
```
![POM_vs_Factory_1](https://github.com/manasamohan16/A49-QA-Automation-Framework/assets/104761482/4800dba3-3280-4833-9072-640e0a7dc1c0)

![POM_vs_Factory_2](https://github.com/manasamohan16/A49-QA-Automation-Framework/assets/104761482/ad8a31af-6088-4653-9647-1dfa8d686393)